lib/src/list_mutations.dart

```diff
@@ -172,14 +176,107 @@ SourceEdit _insertInBlockList(
 
   if (index == list.length) return _appendToBlockList(yamlEdit, list, item);
 
-  final formattedValue = _formatNewBlock(yamlEdit, list, item);
+  var (formattedValue, indent) = _formatNewBlock(yamlEdit, list, item);
 
   final currNode = list.nodes[index];
   final currNodeStart = currNode.span.start.offset;
   final yaml = yamlEdit.toString();
-  final start = yaml.lastIndexOf('\n', currNodeStart) + 1;
 
-  return SourceEdit(start, 0, formattedValue);
+  final currSequenceOffset = yaml.lastIndexOf('-', currNodeStart - 1);
+
+  final (isNested, offset) = _isNestedInBlockList(currSequenceOffset, yaml);
+
+  /// We have to get rid of the left indentation applied by default
+  if (isNested && index == 0) {
+    /// The [insertionIndex] will be equal to the start of
+    /// [currentSequenceOffset] of the element we are inserting before in most
+    /// cases.
+    ///
+    /// Example:
+    ///
+    ///   - - value
+    ///     ^ Inserting before this and we get rid of indent
+    ///
+    /// If not, we need to account for the space between them that is not an
+    /// indent.
+    ///
+    /// Example:
+    ///
+    ///   -   - value
+    ///       ^ Inserting before this and we get rid of indent. But also account
+    ///         for space in between
+    final leftPad = currSequenceOffset - offset;
+    final padding = ' ' * leftPad;
+
+    /// Since we CANT'T/SHOULDN'T manipulate the next element to get rid of the
+    /// space it has, we remove the padding (if any is present) from the indent
+    /// itself.
+    indent = indent.replaceFirst(padding, '');
```

Comment:
Couldn't we use `indent.substring()`...

But also if we made `_formatNewBlock` return `listIndentation` instead of a string, then we can just do `' ' * (listIndentation - leftPad)`, right?

Or am I wrong here?

I'm just wondering, if maybe it's smarter to return integers instead of strings, when we want to communicate indentation. We return an integer from `getListIndentation`.

Honestly, we could probably also just call `getListIndentation(yaml, list);` again, rather than change `_formatNewBlock`, but I don't mind returning both from `_formatNewBlock` :D

Reply:
This actually makes sense. The `_formatNewBlock` can be tweaked and the other reference updated.